Index: book/screen-book/src/main/java/com/rick/screen_book/bestseller_screen/BestsellerEvents.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package com.rick.screen_book.bestseller_screen\r\n\r\nimport com.rick.data_book.nytimes.model.NYBook\r\n\r\nsealed class BestsellerEvents {\r\n\r\n    data class SelectedGenre(val bookGenre: Int): BestsellerEvents()\r\n\r\n    data class OnFavoriteClick(val book: NYBook): BestsellerEvents()\r\n\r\n    data class OnBookClick(val book: NYBook): BestsellerEvents()\r\n\r\n    object OnRemoveFavorite : BestsellerEvents()\r\n\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/book/screen-book/src/main/java/com/rick/screen_book/bestseller_screen/BestsellerEvents.kt b/book/screen-book/src/main/java/com/rick/screen_book/bestseller_screen/BestsellerEvents.kt
--- a/book/screen-book/src/main/java/com/rick/screen_book/bestseller_screen/BestsellerEvents.kt	(revision dbf75b27da7c3ad657e71156cf4e6291e2732b67)
+++ b/book/screen-book/src/main/java/com/rick/screen_book/bestseller_screen/BestsellerEvents.kt	(date 1701682178135)
@@ -4,12 +4,12 @@
 
 sealed class BestsellerEvents {
 
-    data class SelectedGenre(val bookGenre: Int): BestsellerEvents()
+    data class SelectedGenre(val bookGenre: Int) : BestsellerEvents()
 
-    data class OnFavoriteClick(val book: NYBook): BestsellerEvents()
+    data class OnFavoriteClick(val book: NYBook) : BestsellerEvents()
 
-    data class OnBookClick(val book: NYBook): BestsellerEvents()
+    data class OnBookClick(val book: NYBook) : BestsellerEvents()
 
     object OnRemoveFavorite : BestsellerEvents()
 
-}
+}
\ No newline at end of file
Index: movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieAdapter.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieAdapter.kt b/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieAdapter.kt
new file mode 100644
--- /dev/null	(date 1701786538786)
+++ b/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieAdapter.kt	(date 1701786538786)
@@ -0,0 +1,83 @@
+package com.rick.screen_movie.trendingmovie_screen
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.paging.PagingDataAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.rick.data_movie.tmdb.trending_movie.TrendingMovie
+import com.rick.screen_movie.databinding.MovieEntryBinding
+import com.rick.screen_movie.util.provideGlide
+
+class TrendingMovieAdapter(
+    private val binding: MovieEntryBinding,
+    private val onItemClick: (View, TrendingMovie) -> Unit,
+    private val onFavClick: (View, TrendingMovie) -> Unit
+): PagingDataAdapter<TrendingMovie, TrendingMovieViewHolder>(RESULT_COMPARATOR) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TrendingMovieViewHolder {
+        return TrendingMovieViewHolder.create(binding,onItemClick, onFavClick)
+    }
+
+    override fun onBindViewHolder(holder: TrendingMovieViewHolder, position: Int) {
+        val movie = getItem(position)
+        holder.bind(movie!!)
+    }
+
+    companion object {
+        private val RESULT_COMPARATOR = object: DiffUtil.ItemCallback<TrendingMovie>() {
+            override fun areItemsTheSame(oldItem: TrendingMovie, newItem: TrendingMovie): Boolean {
+                return newItem.id == oldItem.id
+            }
+
+            override fun areContentsTheSame(
+                oldItem: TrendingMovie,
+                newItem: TrendingMovie
+            ): Boolean {
+                return newItem == oldItem
+            }
+        }
+    }
+}
+
+class TrendingMovieViewHolder(
+    binding: MovieEntryBinding,
+    private val onItemClick: (View, TrendingMovie) -> Unit,
+    private val onFavClick: (View, TrendingMovie) -> Unit
+): RecyclerView.ViewHolder(binding.root) {
+    private val image = binding.movieImage
+    private val title = binding.movieName
+    private val summary = binding.movieSummary
+    private val favorite = binding.favButton
+
+    private lateinit var movie: TrendingMovie
+
+    fun bind(movie: TrendingMovie) {
+        this.movie = movie
+        if (movie.posterPath.isNotEmpty()) {
+            provideGlide(image, movie.posterPath)
+        }
+        title.text = movie.title
+        summary.text = movie.overview
+    }
+
+    init {
+        binding.movieEntryCardView.setOnClickListener {
+            onItemClick(it, movie)
+        }
+
+        binding.favButton.setOnClickListener {
+            onFavClick(it, movie)
+        }
+    }
+
+    companion object {
+        fun create(
+            binding: MovieEntryBinding,
+            onItemClick: (View, TrendingMovie) -> Unit,
+            onFavClick: (View, TrendingMovie) -> Unit
+        ): TrendingMovieViewHolder {
+            return TrendingMovieViewHolder(binding, onItemClick, onFavClick)
+        }
+    }
+}
\ No newline at end of file
Index: movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieFragment.kt b/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieFragment.kt
new file mode 100644
--- /dev/null	(date 1701682146866)
+++ b/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieFragment.kt	(date 1701682146866)
@@ -0,0 +1,4 @@
+package com.rick.screen_movie.trendingmovie_screen
+
+class TrendingMovieFragment {
+}
\ No newline at end of file
Index: movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieViewModel.kt b/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieViewModel.kt
new file mode 100644
--- /dev/null	(date 1701682141902)
+++ b/movie/screen-movie/src/main/java/com/rick/screen_movie/trendingmovie_screen/TrendingMovieViewModel.kt	(date 1701682141902)
@@ -0,0 +1,4 @@
+package com.rick.screen_movie.trendingmovie_screen
+
+class TrendingMovieViewModel {
+}
\ No newline at end of file
